adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -244,18 +256,20 @@ func (swagger *MgwSwagger) SetXWso2ProductionEndpointMgwSwagger(productionURL st
 	return errors.New("invalid production endpoint")
 }
 
-func (swagger *MgwSwagger) setXWso2PrdoductionEndpoint() error {
-	xWso2APIEndpoints := getXWso2Endpoints(swagger.vendorExtensions, productionEndpoints)
+func (swagger *MgwSwagger) setXWso2ProductionEndpoint() error {
+	xWso2APIEndpoints, err := getXWso2Endpoints(swagger.vendorExtensions, productionEndpoints)
 	if xWso2APIEndpoints != nil && len(xWso2APIEndpoints) > 0 {
 		swagger.productionUrls = xWso2APIEndpoints
-	} else {
+	} else if err != nil {
 		return errors.New("error encountered when extracting endpoints")
 	}
 
 	//resources
 	for i, resource := range swagger.resources {
-		xwso2ResourceEndpoints := getXWso2Endpoints(resource.vendorExtensions, productionEndpoints)
-		if xwso2ResourceEndpoints != nil {
+		xwso2ResourceEndpoints, err := getXWso2Endpoints(resource.vendorExtensions, productionEndpoints)
+		if err != nil {
+			return err
```

Comment:
Shall we log the error perform a continue here?
@Rajith90 wdyt?

Reply:
When it comes to resources, is it okay to continue with the deployment even after we detect a malformed endpoint?